okhttp/src/main/java/okhttp3/UnmodifiableList.java

```diff
@@ -0,0 +1,46 @@
+package okhttp3;
+
+import java.util.Iterator;
+import java.util.List;
+
+public final class UnmodifiableList<T> implements Iterable<T> {
+
+  private final List<T> realList;
+
+  public UnmodifiableList(List<T> realList) {
+    this.realList = realList;
```

Comment:
Missing a copy. You're just caching a mutable list that can be modified after wrapping.